throw-money-app/src/reducks/users/operations.js

```diff
@@ -53,9 +53,45 @@ export const pushRegistUser = (username, email, password, confirmPassword) => {
       .catch((error) => {
         const errorCode = error.code;
         const errorMessage = error.message;
-        alert(`firebase auth ユーザ登録が失敗しました。| errorCode:${errorCode}, errorMessage:${errorMessage}`);
+        alert(`ユーザ登録が失敗しました。| errorCode:${errorCode}, errorMessage:${errorMessage}`);
         return false
       });
   }
+}
+
+// ログイン
+export const login = (email, password) => {
+  return (dispatch) => {
+    // バリデーションチェック
+    if (email === "" || password === "") {

```

Comment:
以下のように条件を記載できます。
```suggestion
    if (!email || !password) {

```

また文字列の囲み方なのですが、`""`ではなく`''`の方が使用されております。
現在はバラバラなので、`''`で統一すると良いです。
フォーマットで自動整形の設定をしてしまうと楽です。
https://qiita.com/mitashun/items/e2f118a9ca7b96b97840

Reply:
以下のコミットにて、試験的にPrettierのフォーマットを適用してみました。
`shift + alt + f `でフォーマット適用後、ログイン機能の動作確認は実施して特に問題ありませんでした。
ただ、ダブルクォーテーション以外のコード部分も自動でインデントが適用されましたが、フォーマットに倣った結果のためそのままでいいと判断しています。(参画するチームのフォーマッタを利用する)

一つ質問になるのですが、`既存ファイル`へフォーマットを適用するためには１ファイルずつ`shift + alt + f `の操作をするしかなさそうでしょうか。（ちなみに`上書き保存時`にはフォーマット適用されませんでした）
「vscode フォーマッター 既存のファイルに適用」のように調べましたが特に見つかりませんでした。

Reply:
>一つ質問になるのですが、既存ファイルへフォーマットを適用するためには１ファイルずつshift + alt + f の操作をするしかなさそうでしょうか。

一気に適用することも可能です。
以下を参考にしてください。
https://chaika.hatenablog.com/entry/2020/06/01/083000
linterとも併用することが多いので、もし余裕があればprettierと併用して使用すると、より実務に近いです。
https://qiita.com/ikngtty/items/4df2e13d2fa1c4c47528

Reply:
試しに紹介記事を参考にlinter/prettierを導入しましたが、以下のエラーとなりました。（一旦べた張りさせて頂きます）
~~~
$ npm run lint-fix

> throw-money-app@0.1.0 lint-fix
> eslint --fix './src/**/*.{js,jsx}' && prettier --write './src/**/*.{js,jsx}'

Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\App.test.js
  4:1   error  'test' is not defined                    no-undef
  5:10  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
  7:3   error  'expect' is not defined                  no-undef

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\component\Button.jsx
  5:34  error  'onClick' is missing in props validation  react/prop-types
  6:14  error  'value' is missing in props validation    react/prop-types

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\component\InputTextForm.jsx
  7:29  error  'name' is missing in props validation       react/prop-types
  8:16  error  'labelName' is missing in props validation  react/prop-types
  9:26  error  'type' is missing in props validation       react/prop-types
  9:44  error  'name' is missing in props validation       react/prop-types
  9:63  error  'value' is missing in props validation      react/prop-types
  9:86  error  'onChange' is missing in props validation   react/prop-types

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\pages\Dashboard.jsx
  2:10  error  'useDispatch' is defined but never used  no-unused-vars

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\pages\Home.jsx
  11:9  error  'userName' is assigned a value but never used  no-unused-vars

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\reducks\users\operations.js
   2:16  error  'goBack' is defined but never used  no-unused-vars
  47:18  error  'result' is defined but never used  no-unused-vars

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\reducks\users\reducers.js
  10:9  error  Parsing error: Unexpected token ..

C:\Users\User\Desktop\MyJSPractice\React-throw-money-app\throw-money-app\src\reportWebVitals.js
  3:5  error  Parsing error: 'import' and 'export' may only appear at the top level

✖ 17 problems (17 errors, 0 warnings)
~~~

試したこと：
①紹介頂いた記事の設定で進めたところ、"Parsing error: unexpected token < (Fatal)"といったエラーがでました。
React用のeslintプラグインが必要なようだったので導入。導入後の状態で、上記のエラーとなっています。
https://teratail.com/questions/140081
https://qiita.com/kjugk/items/b9dfc876e16dbfa4f447

②以下の公式githubより、configurationの値を見比べたりしましたが、なにをどう適応すればいいのかは検討がついていません。
https://github.com/yannickcr/eslint-plugin-react#configuration

ご質問：
①`error  'useDispatch' is defined but never used  no-unused-vars`　
定義したけど使われていない変数についてはエラー無視する設定をどこか(rules?)で追加するような感じで対応できそうでしょうか。
開発を進める上で、変数を先に定義するケースはあると思うので、使われていない変数のエラーに対しては警告レベルに落とせればなと考えています。

②他にも、`error  'onClick' is missing in props validation  react/prop-types`、`Parsing error: Unexpected token ..`や`Parsing error: 'import' and 'export' may only appear at the top level`のエラーも出ていますが、一つ一つエラー内容でググって解決方法を探るしかないでしょうか。
それしかなければ、時間がどれだけかかるかもわからないので、prettierの導入だけにして課題内容を先に行うつもりです。

➂導入後、`npm start`後もコンパイルエラーとなりアプリが立ち上がらなくなりました。
(エラー画面上にはeslintのerrorメッセージが表示されている)
eslintでerror状態が残っていると、「アプリを動かす上では問題ないerrorでも、linterのチェックをクリアしないとアプリを立ち上げることすらもできない」ものなのでしょうか。

以上、お手数おかけしますが宜しくお願い致します。

Reply:
>①error 'useDispatch' is defined but never used no-unused-vars　
定義したけど使われていない変数についてはエラー無視する設定をどこか(rules?)で追加するような感じで対応できそうでしょうか。

はい、可能です。
特定の行に対して設定を無視コメントを記載するか、configファイルに記載して全体に適応させることが可能です。
以下の記事が参考になるかと思います。
https://qiita.com/blajir/items/82127aaa57e4d73a8389
https://qiita.com/mysticatea/items/f523dab04a25f617c87d

>error 'onClick' is missing in props validation react/prop-types、Parsing error: Unexpected token ..やParsing error: 'import' and 'export' may only appear at the top levelのエラーも出ていますが、一つ一つエラー内容でググって解決方法を探るしかないでしょうか。

linter実行の際に、オプションで`--fix`を追加すると自動で補正してくれます。
このコマンドで補正されないものはご自身で修正するしかないですね。
エラーの件もありますし時間がかかるようでしたら、仰る通り今回はprettierだけの導入でも全然良いです。

>➂導入後、npm start後もコンパイルエラーとなりアプリが立ち上がらなくなりました。
(エラー画面上にはeslintのerrorメッセージが表示されている)
eslintでerror状態が残っていると、「アプリを動かす上では問題ないerrorでも、linterのチェックをクリアしないとアプリを立ち上げることすらもできない」ものなのでしょうか。

はい、そうです。
なので、ファイル保存時に自動補完をする設定をしておくのが良いです。
一つ一つ意識して書いていくと、とてもじゃないですがかなりめんどくさいですので。

Reply:
prettierだけの導入とし、prettier整形後のソースをコミットしました。